extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -275,6 +275,7 @@ public void run()
         );
       }
       catch (Exception e) {
+        consumer.close();
```

Comment:
Can you actually move the code below it:
```
      firstRunTime = DateTime.now().plus(ioConfig.getStartDelay());
      scheduledExec.scheduleAtFixedRate(
          buildRunTask(),
          ioConfig.getStartDelay().getMillis(),
          Math.max(ioConfig.getPeriod().getMillis(), MAX_RUN_FREQUENCY_MILLIS),
          TimeUnit.MILLISECONDS
      );

      started = true;
      log.info(
          "Started KafkaSupervisor[%s], first run in [%s], with spec: [%s]",
          dataSource,
          ioConfig.getStartDelay(),
          spec.toString()
      );
``` 
into the try-catch as well? And also you should probably do a (if consumer != null) {consumer.close();} to avoid an NPE